File: src/integration-tests/expecteds/py/pets.py
from typing import List
from typing import NewType
from typing import Union

Ox = NewType("Ox", str)

Horse = NewType("Horse", str)

Donkey = NewType("Donkey", str)

OneOfDonkeyHorseOxP55NQZsj = NewType("OneOfDonkeyHorseOxP55NQZsj", Union[Ox, Horse, Donkey])
"""an array of animals that are good at pulling things. Elaborate enum.
"""
PlowAnimals = NewType("PlowAnimals", List[OneOfDonkeyHorseOxP55NQZsj])

File: src/integration-tests/expecteds/rs/pets.rs
extern crate serde;
extern crate serde_json;
extern crate derive_builder;

use serde::{Serialize, Deserialize};
use derive_builder::Builder;
#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
pub enum Ox {
    #[serde(rename = "Ox")]
    Ox,
}
#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
pub enum Horse {
    #[serde(rename = "Horse")]
    Horse,
}
#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
pub enum Donkey {
    #[serde(rename = "Donkey")]
    Donkey,
}
#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(untagged)]
pub enum OneOfDonkeyHorseOxP55NQZsj {
    Ox(Ox),
    Horse(Horse),
    Donkey(Donkey),
}
/// PlowAnimals
///
/// an array of animals that are good at pulling things. Elaborate enum.
///
pub type PlowAnimals = Vec<OneOfDonkeyHorseOxP55NQZsj>;
